### ISSUE TITLE
Add tabindex to enable navigation flow

### ISSUE THREAD

**Issue by renujain31** [ushahidi/platform#4129]
# Description:
The HTML tabindex attribute is used to manage keyboard focus. It indicates that an element can be focused on, and determines how that focus is handled.
Adding `tabindex="0"` at the places which are not buttons but behave as button element will enable navigation flow and the element would be inserted into the tab order based on its location in the source code.

# Todo:
When we choose `<button>` or `<input>`, keyboard focus and keyboard interaction are handled automatically by the browser. tabindex is helpful when we choose a `<div>` , `<span>` etc as button elements. 

Search for any buttons in the codebase of platform-client which are not button-elements and add `tabindex="0"` in the html element with that class. An easy way to search that might be by searching for `class="button` 

Also make sure that you do not make any changes to the UI of platform and the navigation flow is as expected.

Aha! Link: https://ushahiditeam.aha.io/features/PROD-878

**Comment by hariclerry** [comment]
Hello @renujain31, @rowasc, I'm an outreachy applicant. Can I please work on this?

**Comment by amycruz97** [comment]
can i work on this also

**Comment by KarenEfereyan** [comment]
@renujain31 can I work on this? I'm an outreachy applicant!

**Comment by bukolabisuga** [comment]
Hello @renujain31, I'd love to work on this

**Comment by treasurechristain** [comment]
@Angamanga please let me work on this issue. I'm Amadikwa Joy N. an outreachy applicant

**Comment by rowasc** [comment]
@amycruz97 go ahead. Assigned.

**Comment by amycruz97** [comment]
> @amycruz97 go ahead. Assigned.

pls can u review my other pull request https://github.com/ushahidi/platform-client/pull/1561

**Comment by siggrid** [comment]
@renujain31 @Angamanga 

Sorry, my fault. My suggestion with tabindex=“0” may do more harm than good, if we apply it to links without `href` attributes.

`<a ng-click=function() class=“button”>` is one of the most commonly used non-button element that behaves like a button within the codebase.

What I have found so far:

`<a ng-click=function() class=“button” role=button tabindex=“0”>` receives focus, but doesn’t trigger action when Enter key is pressed 

`<a ng-click=function() class=“button” role=button href=“”>` receives focus, triggers action when Enter key is pressed

**Comment by amycruz97** [comment]
> @renujain31 @Angamanga @rowasc
> 
> Sorry, my fault. My suggestion with tabindex=“0” may do more harm than good, if we apply it to links without `href` attributes.
> 
> `<a ng-click=function() class=“button”>` is one of the most commonly used non-button element that behaves like a button within the codebase.
> 
> What I have found so far:
> 
> `<a ng-click=function() class=“button” role=button tabindex=“0”>` receives focus, but doesn’t trigger action when Enter key is pressed
> 
> `<a ng-click=function() class=“button” role=button href=“”>` receives focus, triggers action when Enter key is pressed

@rowasc please what do you suggest i do, should i go with her second solution

**Comment by siggrid** [comment]
@Angamanga @rowasc 

Have a solution that provides keyboard support for `<a ng-click=function() class=“button”>`

Because this element lacks an href attribute, it is not detected as a link, which normally receives focus and works with Enter.
tabindex=”0” makes the element focusable. But in order to make it interactive we need to add a keyboard event - keydown. If we use role=”button”, then element must work both with Enter and Spacebar. 

Final html template for situations, when buttons are implemented as non-native button elements

`<a role="button" class="button" ng-click="function()" ng-keydown="($event.keyCode === 13 || $event.keyCode === 32) && function()" tabindex="0">Button</a>`

**Comment by Angamanga** [comment]
@siggrid @rowasc @amycruz97  I think when a link has the role of a button rather than a link, we should use the button-element. If that means the styling becomes off, we need to fix the styling in the Pattern-library instead, like we did in this issue: https://github.com/ushahidi/platform-client/pull/1564#issuecomment-714421374.

**Comment by siggrid** [comment]
> @siggrid @rowasc @amycruz97 I think when a link has the role of a button rather than a link, we should use the button-element. If that means the styling becomes off, we need to fix the styling in the Pattern-library instead, like we did in this issue: [ushahidi/platform-client#1564 (comment)](https://github.com/ushahidi/platform-client/pull/1564#issuecomment-714421374).

@Angamanga  Agree. I thought that this must be implemented without changing the markup. As long as native button tags are used, we don’t need to worry about keyboard operability. Might it be applicable to other non-button elements in the codebase as well? For example, [an element that shows/hides a dropdown](https://github.com/ushahidi/platform-client/blob/609914d9649c362ad6800746988b565b1df0197b/app/main/posts/views/filter-by-datasource.html#L21)?

**Comment by Angamanga** [comment]
@siggrid Yes, definitely, eventually we want to improve and make these changes all over the platform 😬 .

**Comment by emmanuelhashy** [comment]
Hi @Angamanga, before I propose to work on this issue, I would like to locate the Pattern-library in the codebase because I have been searching for it for a while now. I think this will give me a better chance of tackling the issue.

**Comment by Angamanga** [comment]
@emmanuelhashy I think most of this is fixed recently, if you want to dig into something more specific, try this task instead: https://github.com/ushahidi/platform/issues/4193

**Comment by emmanuelhashy** [comment]
@Angamanga Alright can you please assign the task to me? Thanks
